hyper/http20/connection.py

```diff
@@ -375,10 +376,23 @@ def receive_frame(self, frame):
             # shutdown and all is well. Otherwise, throw an exception.
             self.close()
 
+            # If an error occured, try to read the error description from
+            # code registry otherwise use the frame's additional data.
             if frame.error_code != 0:
+                err_no = hex(frame.error_code)
+
+                if frame.error_code > 0 and \
+                   frame.error_code < len(H2_ERROR_CODE_REGISTRY):
+                    err_name = H2_ERROR_CODE_REGISTRY[frame.error_code]['Name']
+                    err_description = frame.additional_data or \
+                       H2_ERROR_CODE_REGISTRY[frame.error_code]['Description']
+                else:
+                    err_name = frame.error_code;
+                    err_description = frame.additional_data
```

Comment:
Can we implement all this logic as a lookup function in the registry? It would be nicer for this block of code to be something more like this:

```python
try:
    name, number, description = errors.get_data(frame.error_code)
except ValueError:
    error_string = "Encountered error code %d, extra data %s" % frame.error_code, frame.additional_data
else:
    error_string = "Encountered error %s %s: %s" % (name, number, description)

raise ConnectionError(error_string)
```

It would then be ideal to include a series of tests just to confirm that